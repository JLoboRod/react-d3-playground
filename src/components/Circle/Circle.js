import React from 'react';
import * as d3 from 'd3';

class Circle extends React.Component {
  state = {
    x: this.props.x
  };

  ref = React.createRef();

  componentDidUpdate() {
    d3
      .select(this.ref.current)
      .transition()
      .duration(300)
      .ease(d3.easeCubicInOut)
      .attr('cx', this.props.x)
      .on('end', () => {
        this.setState({
          x: this.props.x
        });
      })
  }

  render() {
    const { x } = this.state;
    const { width, height, onClick } = this.props;

    return (
      <svg style={{ width, height }} onClick={onClick}>
        <circle
          r={10}
          cx={x}
          cy={15}
          ref={this.ref}
        />
      </svg>
    )
  }
}

export default Circle;
