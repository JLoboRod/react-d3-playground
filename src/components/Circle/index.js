import React, { useState } from 'react';

import Circle from './Circle';

export default () => {
  const [isLeft, setIsLeft] = useState(true);

  const handleClick = () => {
    setIsLeft(!isLeft);
  };

  return (
    <Circle
      width={340}
      height={100}
      color='white'
      x={isLeft ? 15 : 300}
      onClick={handleClick}
    />
  );
};