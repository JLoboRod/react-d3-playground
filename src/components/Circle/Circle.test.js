import React from 'react';
import { shallow } from 'enzyme';

import CircleContainer from './';
import Circle from './Circle';

describe('<Circle />', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Circle />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('<CircleContainer />', () => {
  it('should toggle x on click', () => {
    const wrapper = shallow(<CircleContainer />);

    expect(wrapper.props().x).toBe(15);

    wrapper.simulate('click');

    expect(wrapper.props().x).toBe(300);
  });
});
