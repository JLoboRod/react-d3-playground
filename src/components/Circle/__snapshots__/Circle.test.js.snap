// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Circle /> should render without crashing 1`] = `
<svg
  onClick={[Function]}
  style={
    Object {
      "fill": "black",
      "height": 100,
      "width": 340,
    }
  }
>
  <circle
    cx={0}
    cy={15}
    r={10}
  />
</svg>
`;
